msal/src/main/java/com/microsoft/identity/client/AccountAdapter.java

```diff
@@ -54,14 +54,24 @@ private static class GuestAccountFilter implements CacheRecordFilter {
 
         @Override
         public List<ICacheRecord> filter(@NonNull List<ICacheRecord> records) {
+            final String methodTag = GuestAccountFilter.class.getSimpleName() + ":filter";
+
             final List<ICacheRecord> result = new ArrayList<>();
 
             for (final ICacheRecord cacheRecord : records) {
                 final String acctHomeAccountId = cacheRecord.getAccount().getHomeAccountId();
                 final String acctLocalAccountId = cacheRecord.getAccount().getLocalAccountId();
-
-                if (!acctHomeAccountId.contains(acctLocalAccountId)) {
-                    result.add(cacheRecord);
+                try {
+                    if (!acctHomeAccountId.contains(acctLocalAccountId)) {
+                        result.add(cacheRecord);
+                    }
+                } catch (final NullPointerException e) {
+                    if (acctHomeAccountId == null) {
+                        Logger.warn(methodTag, "Home account id is null.");
+                    }
+                    if (acctLocalAccountId == null) {
```

Comment:
Given your code, I think you will get NPE if local account id is null. Only the field on which method is invoked can cause NPE if null (in this case that's home account id)

Reply:
I suggest you re-write your code to avoid NPE in the first place (as opposed to causing NPE and then catching it since this particular NPE is easy to prevent)

Reply:
addressed